rocAL/rocAL/source/bounding_box_graph.cpp

```diff
@@ -147,3 +147,159 @@ void BoundingBoxGraph::update_random_bbox_meta_data(MetaDataBatch *input_meta_da
         input_meta_data->get_bb_labels_batch()[i] = bb_labels;
     }
 }
+
+inline void calculate_ious_for_box(float *ious, BoundingBoxCord box, std::vector<float> anchors)
+{
+    BoundingBoxCord _anchor;
+    _anchor.l = anchors[0];
+    _anchor.t = anchors[1];
+    _anchor.r = anchors[2];
+    _anchor.b = anchors[3];
+    ious[0] = ssd_BBoxIntersectionOverUnion(box, _anchor,true);
+    int best_idx = 0;
+    float best_iou = ious[0];
+
+    for (unsigned int anchor_idx = 1; anchor_idx < (anchors.size() / 4); anchor_idx++)
+    {
+        int m = anchor_idx * 4;
+        BoundingBoxCord _anchor;
+        _anchor.l = anchors[m];
+        _anchor.t = anchors[m + 1];
+        _anchor.r = anchors[m + 2];
+        _anchor.b = anchors[m + 3];
+        float x = ssd_BBoxIntersectionOverUnion(box, _anchor, true);
```

Comment:
copy directly to ious[anchor_idx]. Every extra line of code to copy is at the cost of performance

Reply:
Done